runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/tabs/Rs2Tab.java

```diff
@@ -160,10 +161,34 @@ public static boolean switchToMusicTab() {
 
     public static boolean switchToLogout() {
         if (getCurrentTab() == InterfaceTab.LOGOUT) return true;
-        Widget tab = Microbot.getClient().getWidget(10551341);
+
+        /* Widget Ids - These may change */
+        final int FIXED_CLASSIC_DISPLAY__FIXED_VIEWPORT_OPTIONS_TAB = 35913777;
+        final int RESIZABLE_CLASSIC_DISPLAY__RESIZABLE_VIEWPORT_LOGOUT_ICON = 10551348;
+        final int RESIZABLE_MODERN_DISPLAY__RESIZABLE_VIEWPORT_BOTTOM_LINE_OPTIONS_ICON = 10747938;
+
+        int logout_widget_id;
+
+        try {
+            if (Rs2Widget.getWidget(FIXED_CLASSIC_DISPLAY__FIXED_VIEWPORT_OPTIONS_TAB) != null) {
```

Comment:
try to split out the logic for getting the logout widget and actually logging out.

```
public static boolean switchToLogout() {
    if (getCurrentTab() == InterfaceTab.LOGOUT) return true;

    int logout_widget_id = getLogoutWidgetId();
    
    if (logout_widget_id == 0) return false;

    Widget tab = Microbot.getClient().getWidget(logout_widget_id);
    if (tab == null) return false;
    
    Microbot.getMouse().click(tab.getBounds());
    sleep(200, 600);
    
    return getCurrentTab() == InterfaceTab.LOGOUT;
}

private static int getLogoutWidgetId() {
    try {
        if (Rs2Widget.getWidget(FIXED_CLASSIC_DISPLAY__FIXED_VIEWPORT_OPTIONS_TAB) != null) {
            return FIXED_CLASSIC_DISPLAY__FIXED_VIEWPORT_OPTIONS_TAB;
        } else if (Rs2Widget.getWidget(RESIZABLE_CLASSIC_DISPLAY__RESIZABLE_VIEWPORT_LOGOUT_ICON) != null) {
            return RESIZABLE_CLASSIC_DISPLAY__RESIZABLE_VIEWPORT_LOGOUT_ICON;
        } else if (Rs2Widget.getWidget(RESIZABLE_MODERN_DISPLAY__RESIZABLE_VIEWPORT_BOTTOM_LINE_OPTIONS_ICON) != null) {
            return RESIZABLE_MODERN_DISPLAY__RESIZABLE_VIEWPORT_BOTTOM_LINE_OPTIONS_ICON;
        }
    } catch (Exception ignored) {
        // Rs2Widget.getWidget returns null if the game isn't finished loading
        // Consider logging the exception for debugging purposes
    }

    return 0;
}
```

Reply:
Could you elaborate, this should be just switching to the required tab. I only extended the original method to account for all three display settings.

Reply:
SwitchToLogout and getting the logout widget are two seperate tasks, and it is better to split them up. code easier to read and maintain and won't violate the SRP principle.